backend/src/main/java/com/htecgroup/skynest/controller/UserController.java

```diff
@@ -18,11 +26,83 @@
 @RequestMapping("/users")
 @AllArgsConstructor
 @Log4j2
+@Tag(name = "User API", description = "Operations to manipulate user")
 public class UserController {
 
   private UserService userService;
   private ModelMapper modelMapper;
 
+  @Operation(summary = "Register new user")
+  @ApiResponses(
+      value = {
+        @ApiResponse(
+            responseCode = "200",
+            description = "Registration successful",
+            content = {
+              @Content(
+                  mediaType = "application/json",
+                  schema = @Schema(implementation = UserResponse.class),
+                  examples = {
+                    @ExampleObject(
+                        value =
+                            "{\"id\": \"a6fd6d95-0a60-43ff-961f-2b9b2ff72f95\","
+                                + " \"email\": \"username@gmail.com\","
+                                + "  \"name\": \"Name\","
+                                + "  \"surname\": \"Surname\","
+                                + "  \"phoneNumber\": \"38166575757\","
+                                + "  \"address\": \"Local address\"}")
+                  })
+            }),
+        @ApiResponse(
+            responseCode = "400",
+            description = "Input fields are not in valid format",
+            content = {
+              @Content(
+                  mediaType = "application/json",
+                  schema = @Schema(implementation = ErrorMessage.class),
+                  examples = {
+                    @ExampleObject(
+                        value =
+                            "{\"messages\":[\"password format not valid\","
+                                + " \"name cannot be null or empty\","
+                                + " \"surname cannot be null or empty\","
+                                + " \"email format is not valid\","
+                                + " \"phoneNumber format not valid\"],"
+                                + " \"status\": \"400\","
+                                + " \"timestamp\": \"2022-06-03 16:18:12\"}")
+                  })
+            }),
+        @ApiResponse(
+            responseCode = "409",
+            description = "Email or phone number already in use",
+            content = {
+              @Content(
+                  mediaType = "application/json",
+                  schema = @Schema(implementation = ErrorMessage.class),
+                  examples = {
+                    @ExampleObject(
+                        value =
+                            "{\"messages\":[\"This email is already in use\"],"
+                                + " \"status\": \"400\","
+                                + " \"timestamp\": \"2022-06-03 16:18:12\"}")
+                  })
+            })
+      })
```

Comment:
You don't expect error 500 here? :P :D

Reply:
Definitely :D Also I saw in example that I missed status code.. Fixed it

Reply:
maybe for consistency use the same format as the other error 500 desc